crates/router/src/connector/zen/transformers.rs

```diff
@@ -0,0 +1,393 @@
+}
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct ZenMerchantActionData {
+    redirect_url: url::Url,
+}
+
+impl<F, T>
+    TryFrom<types::ResponseRouterData<F, ZenPaymentsResponse, T, types::PaymentsResponseData>>
+    for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<F, ZenPaymentsResponse, T, types::PaymentsResponseData>,
+    ) -> Result<Self, Self::Error> {
+        let redirection_data_action = item.response.merchant_action.map(|merchant_action| {
+            (
+                services::RedirectForm::from((merchant_action.data.redirect_url, Method::Get)),
+                merchant_action.action,
+            )
+        });
+        let (redirection_data, action) = match redirection_data_action {
+            Some((redirect_form, action)) => (Some(redirect_form), Some(action)),
+            None => (None, None),
+        };
+
+        Ok(Self {
+            status: enums::AttemptStatus::foreign_try_from((item.response.status, action))?,
+            response: Ok(types::PaymentsResponseData::TransactionResponse {
+                resource_id: types::ResponseId::ConnectorTransactionId(item.response.id),
+                redirection_data,
+                mandate_reference: None,
+                connector_metadata: None,
+            }),
+            ..item.data
+        })
+    }
+}
+
+#[derive(Default, Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ZenRefundRequest {
+    amount: String,
+    transaction_id: String,
+    currency: enums::Currency,
+    merchant_transaction_id: String,
+}
+
+impl<F> TryFrom<&types::RefundsRouterData<F>> for ZenRefundRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::RefundsRouterData<F>) -> Result<Self, Self::Error> {
+        Ok(Self {
+            amount: utils::to_currency_base_unit(
+                item.request.refund_amount,
+                item.request.currency,
+            )?,
+            transaction_id: item.request.connector_transaction_id.clone(),
+            currency: item.request.currency,
+            merchant_transaction_id: item.request.refund_id.clone(),
+        })
+    }
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Serialize, Default, Deserialize, Clone)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum RefundStatus {
+    Authorized,
+    Accepted,
+    #[default]
+    Pending,
+    Rejected,
+}
+
+impl From<RefundStatus> for enums::RefundStatus {
+    fn from(item: RefundStatus) -> Self {
+        match item {
+            RefundStatus::Accepted => Self::Success,
+            RefundStatus::Pending | RefundStatus::Authorized => Self::Pending,
+            RefundStatus::Rejected => Self::Failure,
+        }
+    }
+}
+
+//TODO: Fill the struct with respective fields
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
```

Comment:
For all the struct and enum remove Clone, Eq and PartialEq. 
For request use Serialize and response use Deserialize

Please verify this in all the struct and enums in this transformers file